source/solvers/cahn_hilliard.cc

```diff
@@ -522,6 +532,135 @@ CahnHilliard<dim>::postprocess(bool first_iteration)
+                this->present_solution,
+                *multiphysics->get_block_time_average_solution(
+                  PhysicsID::fluid_dynamics));
+            }
+          else
+            {
+              position_and_velocity =
+                calculate_barycenter(this->present_solution,
+                                     *multiphysics->get_block_solution(
+                                       PhysicsID::fluid_dynamics));
+            }
+        }
+      else
+        {
+          if (this->simulation_parameters.multiphysics
+                .use_time_average_velocity_field &&
+              simulation_control->get_current_time() >
+                this->simulation_parameters.post_processing.initial_time)
+            {
+              position_and_velocity =
+                calculate_barycenter(this->present_solution,
+                                     *multiphysics->get_time_average_solution(
+                                       PhysicsID::fluid_dynamics));
+            }
+          else
+            {
+              position_and_velocity =
+                calculate_barycenter(this->present_solution,
+                                     *multiphysics->get_solution(
+                                       PhysicsID::fluid_dynamics));
+            }
+        }
+      if (this_mpi_process == 0)
+        {
+          if (simulation_parameters.post_processing.verbosity ==
+              Parameters::Verbosity::verbose)
+            {
+              std::cout << std::endl;
+              std::string independent_column_names = "time";
+
+              std::vector<std::string> dependent_column_names;
+              dependent_column_names.push_back("x_cahn_hilliard");
+              dependent_column_names.push_back("y_cahn_hilliard");
+              if constexpr (dim == 3)
+                dependent_column_names.push_back("z_cahn_hilliard");
+              dependent_column_names.push_back("vx_cahn_hilliard");
+              dependent_column_names.push_back("vy_cahn_hilliard");
+              if constexpr (dim == 3)
+                dependent_column_names.push_back("vz_cahn_hilliard");
+
+              std::vector<Tensor<1, dim>> position_vector{
+                position_and_velocity.first};
+              std::vector<Tensor<1, dim>> velocity_vector{
+                position_and_velocity.second};
+
+              std::vector<std::vector<Tensor<1, dim>>>
+                position_and_velocity_vectors{position_vector, velocity_vector};
+
+              std::vector<double> time = {
+                this->simulation_control->get_current_time()};
+
+              TableHandler table = make_table_scalars_tensors(
+                time,
+                independent_column_names,
+                position_and_velocity_vectors,
+                dependent_column_names,
+                this->simulation_parameters.simulation_control.log_precision);
+
+              std::cout << "+------------------------------------------+"
```

Comment:
You could use  announce_string from utilities.h here. It is merely cosmetic, but eventually someone will need to change it if you do not implement in this version.